docs/DeveloperGuide.md

```diff
@@ -175,7 +219,20 @@ The second sequence diagram given below shows the detailed interaction that aces
 
 ![CalcReffSequence](diagrams/CalcReffSequenceDiagram.png)
 
-<small><i>Figure 12</i></small>  
+<small><i>Figure 12</i></small> 
+
+### Implementation Considerations For Circuit Commands (Dash) <a name="impl-cons-dash"></a>
+This section describes the methods taken into consideration while implementing the Circuit Commands.
+
+#### Rationale Behind The Implementation of Circuit Template <a name="rationale-circuit-dash"></a>
+This section details the rationale for implimenting the circuit template as described in the prior sections. Selecting a way to build circuits is the primary purpose of the system and hence a way to design the circuit is required. Since the system is designed primarily as a command line interface, creating complex customizable circuits using a graphical interface is beyond the scope of this system. To implement a circuit four different `template` classes are used, which inherit from a superclass `template`. The four different templates are also presented as simple logos to show how the templates would look like as real life circuits. Furthermore, the function to add components to the circuit calculates the effective value with the components that are already present. For example when a resistor is added in parallel, the effective resistance of the aforementioned resistor is added to the effective resistance of the resistor already present in the circuit template. The two resistors are then replaced by one resistor with the same effective resistance. Similarly, other components are also added and abstracted into a single component of the matching component type.
+
+#### Alternatives Considered <a name="alt-circuit-dash"></a>
+This section illustrates the other alternatives considered for implementing the Circuit Template.
+
+* Linked List: A linked list implementation involves using each circuit component as a node and the connections between the nodes as wires. The head of the linked list is the voltage source and the a loop is present from the tail node number 2 (i.e index 1). However, such an approach allows users to add any component even when certain circuits are not usually built with those components. To fulfill the purpose of the system to cater to the needs of CG111 students this approach is avoided. The circuit templates on the other hand allow the user to distiguish between different circuits and do so effectively.     
```

Comment:
> However, such an approach allows users to add... 

I'm not quite sure whether this would be an issue since we could constrain the user from adding certain components to the circuit. Perhaps you could say that a Linked List would not be able to simulate a proper closed circuit-like system, therefore was not chosen...?

Reply:
My point here is that if we did choose a linked list structure then we would have no templates so we just add the components directly to the linked list. Of course, we could block users from adding certain components by checking the components present in the linked list. For some reason, I feel like a linked list interpretation would have worked. Any specific reason why it wouldn't be able to simulate a circuit system?

Reply:
I think we are all forgetting the GUI constraint here. Any dynamic visualisation would dangerously be part of GUI, and making everything static (similarly for Boolean) cuts that issue out.